pkg/activator/handler/handler.go

```diff
@@ -55,26 +60,64 @@ func (a *ActivationHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	start := time.Now()
 	revID := activator.RevisionID{Namespace: namespace, Name: name}
 
+	ctx := r.Context()
+	tracerRef, err := a.TRGetter(ctx)
+	var tracer *zipkin.Tracer
+	var activateSpan zipkin.Span
+	if tracerRef == nil {
+		if enabled, ok := config.TracingEnabledForContext(ctx); enabled {
```

Comment:
In general it feels that some of this logic belongs elsewhere and here we should just do
```
x := a.beginTrace()
defer x.EndTrace()
// or
defer a.beginTrace().EndTrace()
```
Is tracing on/off on per-service or per-deployment?

Reply:
I agree guarding against this doesnt feel right. The reasoning is for the failure case (failed to get a tracer for some reason) more than the configured on/off case. There was a comment earlier about we should not fail a request due to a tracer failing which makes some sense, but the result is we have to guard against tracer failures all over like this.

The options I see are:
- Continue this type of guarding
- Fail requests if we cant get a tracer. I am not sure theres much surface area for this TBH (even invalid config cases we can generate a noop tracer) so it may be worth trying this until an issue comes up.
- We can generate a noop tracer on error in tracer cache so TracerRefGetter can always return a ref and not error.

Thoughts?

Reply:
I think `noop` tracer that just does nothing would really streamline the readability here.